prometheus/testutil/testutil.go

```diff
@@ -138,20 +138,13 @@ func GatherAndCompare(g prometheus.Gatherer, expected io.Reader, metricNames ...
 	}
 	want := internal.NormalizeMetricFamilies(wantRaw)
 
-	if len(got) != len(want) {
-		return notMatchingError(got, want)
-	}
-	for i := range got {
-		if got[i].String() != want[i].String() {
-			return notMatchingError(got, want)
-		}
-	}
-	return nil
+	return compare(got, want)
 }
 
-// notMatchingError encodes both provided slices of metric families into the
-// text format and creates a readable error message from the result.
-func notMatchingError(got, want []*dto.MetricFamily) error {
+// compare encodes both provided slices of metric families into the
+// text format compares their string message and returns an error if they do not match.
+// In case of error, prints a readable error message from the result.
```

Comment:
This sounds like it would call `fmt.Println` or something. Perhaps just mention that the returned error will contain the encoded text of both, got and want.